### PR TITLE
Feature/save default meal

### PR DESCRIPTION
**Pull request by maximest-pierre**
*Fixes # .*

### Changes proposed in this pull request:

* Add a json filed inside the model
* Add a way to save the default by day

### Status

- [x] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### Deployment notes and migration

- [X] Migration needed

Fill me in ...

@savoirfairelinux/mll

### CODE REVIEW

File: django/santropolFeast/member/templates/forms/steps/dietary_restriction.html [hunk 1]
```diff
@@ -27,10 +28,408 @@ <h4 class="ui dividing header">{{ _('Meals default') }}</h4>
...
+        </tr>
+        <tr>
+          <td></td>
+          <td>{% trans 'Compote' %}</td>
+          <td>{{wizard.form.compote_sunday_quantity}}</td>
+        </tr>
+      </tbody>
+    </table>
+</div>
+</div>
```
Comment (manumilou) [inline django/santropolFeast/member/templates/forms/steps/dietary_restriction.html]: Maybe a for loop? :-)

File: django/santropolFeast/member/views.py [hunk 2]
```diff
@@ -109,7 +109,199 @@ def save(self):
...
+                                   ), 'green_salad_sunday_quantity':
+                                 dietary_restriction.cleaned_data.get(
+                                    'green_salad_sunday_quantity'
+                                   ), 'pudding_sunday_quantity':
+                                 dietary_restriction.cleaned_data.get(
+                                    'pudding_sunday_quantity'
+                                   ), 'compote_sunday_quantity':
+                                 dietary_restriction.cleaned_data.get(
+                                    'compote_sunday_quantity'
+                                   )
```
Comment (manumilou) [inline django/santropolFeast/member/views.py]: This is annoying, please use a for loop

File: django/santropolFeast/member/forms.py [hunk 3]
```diff
@@ -111,6 +138,33 @@ class ClientRestrictionsInformation(forms.Form):
...
+        queryset=Ingredient.objects.all(),
+        required=False,
+        widget=forms.SelectMultiple(
+            attrs={'class': 'ui dropdown search'}
+        )
+    )
+
+    dish_to_avoid = forms.ModelMultipleChoiceField(
+        label=_("Dish To Avoid"),
+        queryset=Ingredient.objects.all(),
```
Comment (lamontfr) [inline django/santropolFeast/member/forms.py]: I think it should be Component, not Ingredient

Comment (maximest-pierre) [inline django/santropolFeast/member/forms.py]: I am going to fix it right now but yeah it should be Component.

**Comment by maximest-pierre** [comment]
{% load i18n %}
{% for day in day_of_week %}
<div class="ui styled fluid accordion">
  <div class="title">
    <i class="dropdown icon"></i>
    {% trans day %}
  </div>
  <div class="content">
    <table class="ui compact table">
      <thead>
        <tr>
          <th>{% trans 'Size' %}</th>
          <th>{% trans 'Component' %}</th>
          <th>{% trans 'Quantity' %}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>{{wizard.form.size_{{day}}}}</td>
          <td>{% trans 'Main Dish' %}</td>
          <td>{{wizard.form.main_dish_{{day}}_quantity}}</td>
        </tr>
      {% for meal in meals %}
        <tr>
          <td></td>
          <td>{% trans {{meal}}}</td>
          <td>{{wizard.form.{{meal}}_{{day}}_quantity}}</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>

**Comment by maximest-pierre** [comment]
https://gist.github.com/maximest-pierre/29a04e3110d089f46ce9762c34c4faf9

**Comment by lamontfr** [comment]
@maximest-pierre   As discussed on Friday July 7th, here is an example of how to use a custom tag to simplify the template (django/santropolFeast/member/templates/forms/meal_default.html). The dish_day custom tag can use any python code to return the desired string that will be expanded in the page generated by the template.

**FILE django/santropolFeast/meal/templatetags/custom_tags.py**

```python
from django import template

register = template.Library()

@register.simple_tag(takes_context=True, name='cell')
def dish_day(context, dish, day):
    # print(context)
    return context['wizard']["form"][dish+"_"+day+"_quantity"]
```

**Run this script to see how it works :**
**FILE django/santropolFeast/FRL_ctt.py**

```python
# run in :  $ python manage.py shell
#           >>> exec(open('FRL_ctt.py').read())

from django import template
from meal.templatetags import custom_tags

cont = template.Context({"wizard": {"form":
                                    {"pudding_Monday_quantity": 10,
                                     "pudding_Tuesday_quantity": 20,
                                     "pudding_Wednesday_quantity": 30,
                                     "pudding_Thursday_quantity": 40,
                                     "pudding_Friday_quantity": 50,
                                     "pudding_Saturday_quantity": 60,
                                     "pudding_Sunday_quantity": 70,
                                     "dessert_Monday_quantity": 100,
                                     "dessert_Tuesday_quantity": 200,
                                     "dessert_Wednesday_quantity": 300,
                                     "dessert_Thursday_quantity": 400,
                                     "dessert_Friday_quantity": 500,
                                     "dessert_Saturday_quantity": 600,
                                     "dessert_Sunday_quantity": 700}
                                   },
                         "dishes":["pudding", "dessert"],
                         "day_of_week":
                         ("Monday", "Tuesday", "Wednesday",
                          "Thursday", "Friday", "Saturday", "Sunday")
                        })

print(template.
      Template(
          """
          {% load custom_tags %}
          {% for day in day_of_week %}
          {{ day }}
          {% for dish in dishes %}
          <tr>
          <td></td>
          <td>{{ dish }}</td>
          <td>{% cell dish day %}</td>
          </tr>
          {% endfor %} {% endfor %}
          """).
      render(cont))
```

**OUTPUT**
```python
f:\fow\santropol-feast\django\santropolFeast>python manage.py
Python 3.4.4 (v3.4.4:737efcadf5a6, Dec 20 2015, 19:28:18) [MSC v.1600 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> exec(open('FRL_ctt.py').read())



          Monday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>10</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>100</td>
          </tr>

          Tuesday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>20</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>200</td>
          </tr>

          Wednesday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>30</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>300</td>
          </tr>

          Thursday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>40</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>400</td>
          </tr>

          Friday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>50</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>500</td>
          </tr>

          Saturday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>60</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>600</td>
          </tr>

          Sunday

          <tr>
          <td></td>
          <td>pudding</td>
          <td>70</td>
          </tr>

          <tr>
          <td></td>
          <td>dessert</td>
          <td>700</td>
          </tr>


>>>
```

**Comment by maximest-pierre** [comment]
I was messing around with custom tag and it seems like the best way to handle this problem I will update this pr tomorrow night.

**Comment by maximest-pierre** [comment]
thanks @lamontfr  It works. I am updating the pr and squashing some little bugs.

**Pull request merged**